addons/purchase/models/purchase.py

```diff
@@ -1068,11 +1068,16 @@ def _compute_qty_received_method(self):
 
     @api.depends('qty_received_method', 'qty_received_manual')
     def _compute_qty_received(self):
-        for line in self:
-            if line.qty_received_method == 'manual':
-                line.qty_received = line.qty_received_manual or 0.0
-            else:
-                line.qty_received = 0.0
+        lines_manual = self.filtered(lambda line: (
+            line.qty_received_method == 'manual'
+            and line.qty_received != line.qty_received_manual
+        ))
+        for line in lines_manual:
+            line.qty_received = line.qty_received_manual or 0.0
+
+        self.filtered(lambda line: line.qty_received_method != 'manual').write({
+            'qty_received': 0.0,
+        })
```

Comment:
OMG please don't use `write` to assign a field in a compute method.
```suggestion
        self.filtered(lambda line: line.qty_received_method != 'manual').qty_received = 0.0
```